File: backend/src/main/java/com/back/domain/item/item/dto/ItemCreateRequest.java
package com.back.domain.item.item.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import org.springframework.web.multipart.MultipartFile;

import java.time.LocalDate;

public record ItemCreateRequest(
        @NotNull Long categoryId,
        @NotBlank String name,
        String imgUrl,
        MultipartFile image,
        LocalDate startDate,
        @NotBlank String cycleDays
) {
    public LocalDate resolvedStartDate() {
        return (startDate == null) ? LocalDate.now() : startDate;
    }
}

File: backend/src/main/java/com/back/domain/item/item/dto/ItemCreateRequest.kt
package com.back.domain.item.item.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import org.springframework.web.multipart.MultipartFile
import java.time.LocalDate

data class ItemCreateRequest(
    @JvmField val categoryId: Long,

    @field:NotBlank
    @JvmField val name: String,

    @JvmField val imgUrl: String?,
    @JvmField val image: MultipartFile?,
    val startDate: LocalDate?,

    @field:NotBlank
    @JvmField val cycleDays: String
) {
    fun resolvedStartDate(): LocalDate {
        return startDate ?: LocalDate.now()
    }
}